lib/saluki-components/src/transforms/dogstatsd_prefix_filter/mod.rs

```diff
@@ -290,6 +290,29 @@ mod tests {
         assert_eq!(new_metric.context().name().clone().into_owned(), "foo".to_string());
     }
 
+    #[test]
+    fn test_metric_blocklist_with_metric_prefix() {
+        let filter = DogstatsDPrefixFilter {
+            metric_prefix: "foo".to_string(),
+            metric_prefix_blacklist: vec![],
+            blocklist: Blocklist::new(&["foobar".to_string(), "test".to_string()], false),
+        };
+        let context = Context::from_static_parts("bar", &[]);
+        let metric = Metric::gauge(context, 1.0);
+        let new_metric = filter.enrich_metric(metric);
+        assert!(new_metric.is_none());
+
+        let filter = DogstatsDPrefixFilter {
```

Comment:
While testing/looking at the agent code, I saw that a metric won't be blocked if the name is the same as a blocked metric prefix but not in the block list 🤔 . (`foo` prefix is not added to make `foofoo` since `foo` in `metric_prefix_blacklist` but `foo` is not in `blocklist`)

Reply:
That makes sense to me, but admittedly is why I think the Agent code (and this code as a result) is a little confusing... because it's coupling the logic for prefixing and the logic for the blocklist stuff into the same area.